src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -25,30 +24,73 @@ export const MultiSelectDialog = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectDialogProps>>(
     (props, ref) => {
       const {
+        helperText,
         id,
         isBlockElement,
         isDefaultOpen,
+        isSearchable,
+        defaultItemsVisible,
         items,
-        labelText,
+        listOverflow,
+        buttonText,
+        closeOnBlur,
         onApply,
         onChange,
         onClear,
         onMixedStateChange,
         selectedItems,
+        showLabel,
+        showAll,
         width,
         ...rest
       } = props;
 
-      // Use NYPL Breakpoints hook to check screen size
-      const { isLargerThanMobile } = useNYPLBreakpoints();
-
       // Control the open or closed state of the MultiSelect.
-      const [isOpen, setIsOpen] = useState(isDefaultOpen);
+      const [isOpen] = useState(isDefaultOpen);
+      const [itemsList, setItemsList] = useState(items);
+      const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+      const [viewAllLabel, setViewAllLabel] = useState("");
+      const [listHeight, setListHeight] = useState("");
+      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+      const [defaultCheckboxes, setDefaultCheckboxes] = useState("none");
+      const [defaultChildCheckboxes, setDefaultChildCheckboxes] =
+        useState("none");
+      //const [prevIsOpen, setPrevIsOpen] = React.useState(isDropDownOpen);
+
+      React.useEffect(() => {
+        const calculateListHeight = () => {
+          if (listOverflow === "scroll") {
+            setViewAllLabel("View all");
+            setListHeight("260px");
+            setListItemsCount(defaultItemsVisible);
```

Comment:
`listOverflow` does not get updated in this component based on state so you can set these as default when they are declared rather than here.
For example, for `setViewAllLabel("View all");`, this would be better on line 52 as
```
const [viewAllLabel, setViewAllLabel] = useState<string>(listOverflow === "scroll" ? "View all" : "");
```

The same is true for `setListHeight`.